src/SOS/extensions/hostcoreclr.cpp

```diff
@@ -246,10 +260,54 @@ static void AddFilesFromDirectoryToTpaList(const char* directory, std::string& t
     }
 }
 
+static std::string GetTpaListForRuntimeVersion(
+    const std::string& sosModuleDirectory,
+    const std::string& hostRuntimeDirectory,
+    const RuntimeVersion& hostRuntimeVersion)
+{
+    std::string tpaList;
+    const char* directory = sosModuleDirectory.c_str();
+
+    // TODO: Two important questions. Which file is greater version and is 
+    //       this an exhaustive list.
+    if (hostRuntimeVersion.Major < 5)
+    {
+        AddFileToTpaList(directory, "System.Collections.Immutable.dll", tpaList);
+        AddFileToTpaList(directory, "System.Reflection.Metadata.dll", tpaList);
+        AddFileToTpaList(directory, "System.Runtime.CompilerServices.Unsafe.dll", tpaList);
+    }
+
+    // AddFileToTpaList(directory, "Microsoft.Bcl.AsyncInterfaces.dll", tpaList);
```

Comment:
Do we ever need to add these assemblies? Are you going to leave these commented out lines?

Reply:
I put them there to leave them up for discussion and to ask Vitek why weren't those needed. I'll ask you on monday if we know in what circumstances things are needed.